### PR TITLE
Upgrade to Akka 2.6.1, akka-http 10.1.11 and Scala 2.12.10

### PR DESCRIPTION
**Pull request by dasormeter**
Akka 2.6.1 Upgrade Changes
- Materializer in place of ActorMaterializer
- Source.future instead of Source.fromFuture
- The Scheduler.schedule method has been deprecated in favor of selecting scheduleWithFixedDelay or scheduleAtFixedRate
- onDownstreamFinish(cause: Throwable)
- remove unneeded usages of ActorAttributes.supervisionStrategy(...) in place of ActorMaterializerSettings.withSupervisionStrategy

See https://doc.akka.io/docs/akka/current/project/migration-guide-2.5.x-2.6.x.html

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/scala/bindings-akka-testing/src/main/scala/com/digitalasset/ledger/client/testing/AkkaTest.scala [hunk 1]
```diff
@@ -35,8 +35,8 @@ trait AkkaTest extends BeforeAndAfterAll with LazyLogging { self: Suite =>
   protected implicit val scheduler: Scheduler = system.scheduler
   protected implicit val schedulerService: ScheduledExecutorService =
     Executors.newSingleThreadScheduledExecutor()
-  protected implicit val materializer: ActorMaterializer = ActorMaterializer(
-    ActorMaterializerSettings(system).withSupervisionStrategy(decider(system)))
+  protected implicit val materializer: Materializer = Materializer(system)
+  ActorAttributes.supervisionStrategy(decider(system))
```
Comment (SamirTalwar) [inline language-support/scala/bindings-akka-testing/src/main/scala/com/digitalasset/ledger/client/testing/AkkaTest.scala]: Looks like [`ActorAttributes.supervisionStrategy`, etc. are pure functions](https://github.com/akka/akka/blob/v2.6.1/akka-stream/src/main/scala/akka/stream/Attributes.scala#555), so this doesn't do anything any more. If everything is still working with this change, we can probably delete it (and `decider`) entirely.

I don't know why this was done in the first place, though, so I'd be wary. If we want to make sure this still works, we need to set it on the stream itself:

```scala
Source(…)
  .blah
  .toMat(sink)(Keep.right)
  .withAttributes(ActorAttributes.supervisionStrategy(decider(system))
```

File: ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala [hunk 2]
```diff
@@ -37,12 +37,11 @@ object ReferenceServer extends App {
...
-    ActorMaterializerSettings(system)
-      .withSupervisionStrategy { e =>
-        logger.error(s"Supervision caught exception: $e")
-        Supervision.Stop
-      })
+  implicit val materializer: Materializer = Materializer(system)
+  ActorAttributes.supervisionStrategy { e =>
+    logger.error(s"Supervision caught exception: $e")
+    Supervision.Stop
+  }
```
Comment (SamirTalwar) [inline ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala]: See comment above.

File: language-support/scala/bindings-akka-testing/src/main/scala/com/digitalasset/ledger/client/testing/AkkaTest.scala (hunk 1 above)
Comment (dasormeter) [reply language-support/scala/bindings-akka-testing/src/main/scala/com/digitalasset/ledger/client/testing/AkkaTest.scala]: removed

File: ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala (hunk 2 above)
Comment (dasormeter) [reply ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala]: removed

**Comment by dasormeter** [comment]
> /azp run

mac build is 3hrs in and seems to be dead : kicking off a new run

**Pull request merged**